### ISSUE TITLE
GeoIP Update Fail

### ISSUE THREAD

**Issue by p-yukusai** [winsiderss/systeminformer#1488]
### Brief description of your issue

System Informer fails to download the GeoIP database, thus making the country column in the network tab unusable. <br>
![image](https://user-images.githubusercontent.com/85849071/202047586-ea72994c-31ce-4c74-9687-18a5a9b580fa.png)

### Steps to reproduce (optional)

Select "GeoIP database update" under Tools -> Network Tools

### Expected behavior (optional)

Successful GeoIP download or indications for manually achieving the same.

### Environment (optional)

```shell
System Informer 3.0.5520
Windows 11
```

**Comment by rpodric** [comment]
So does the last nightly version of Process Hacker though (as of today--it was working as of a few months ago when I tried it last) suggesting that it's the server, not the client. Or the method of contact that both programs use (which is likely the same) needs updating.

**Comment by enervated** [comment]
The GeoIP database update queries wj32.org, which is down. I'm guessing they're working on hosting it elsewhere as part of the Process Hacker -> System Informer migration. Guessing.

**Comment by jrmoore** [comment]
In the meantime, put the database file manually in `%AppData%\SystemInformer`.

The file attached to this comment should contain the latest version at the time of writing, extract it there.

[GeoLite2-Country_20221115.tar.gz](https://github.com/winsiderss/systeminformer/files/10034386/GeoLite2-Country_20221115.tar.gz)

**Comment by Biswa96** [comment]
If you can provide an official direct link to download the file that would be useful. Random files in internet is not so trustworthy.

**Comment by jrmoore** [comment]
> If you can provide an official direct link to download the file that would be useful. Random files in internet is not so trustworthy.

I see! The free GeoLite2 requires a parameter key for direct download links, so I can't link directly to it; but in their site one can create an account to download the current version of the databases.

https://dev.maxmind.com/geoip/geolite2-free-geolocation-data?lang=en

The one for countries is the one Process Hacker / System Informer uses, in MMDB format:

![image](https://user-images.githubusercontent.com/311922/202529202-6499f1bc-8ed9-47b5-993e-d0283a13fa7e.png)

**Comment by rpodric** [comment]
This is interesting. The SI data dir does actually need to be SystemInformer (as you point out) and not System Informer, even if your program dir for SI is System Informer. My directory didn't exist at all, so I had to create it manually. Initially I made it "System Informer" as the dir I have SI in is two words (just like Process Hacker and its data dir were). But when it's two words, the db can't be found. Anyway, thanks for the file and explanation of the problem.

**Comment by enervated** [comment]
uckin ell

**Comment by jrmoore** [comment]
I noticed the folder under `%AppData%` lacking a separating space too @rpodric, I just switched from Process Hacker today so the current nightly is the first version I've installed. The installer defaulted to `%ProgramFiles%\SystemInformer` as well, so at first I thought the program was actually named or stylized like that, but after opening it and everywhere else it reads System Installer.

I had the folder already because I usually change the default settings on first launch (dark theme, check for digital signatures, hiding toolbar, collapse services, etc.) and they get stored there.

**Comment by p-yukusai** [comment]
I can confirm that jmoore's solution works as intended, hope this can be done in-app soon enough 😄

**Comment by Vasilich** [comment]
i have used kinda portable version of PH, and have expected that SystemInformer will also support portable mode. And while it still supports keeping config file in its own folder, the maxmind database path seems to disappeared from config. At least when i try to put the setting back n config file, it disappears after SystemInformer start.
```XML
<setting name="ProcessHacker.NetworkTools.GeoDbPath">.\plugins\maxminddb\GeoLite2-Country.mmdb</setting>
```
Is there any chance to get this setting back?

**Comment by pasha-zzz** [comment]
http://geolite2.6te.net/GeoLite2-Country.7z

Daily updated ~at 0:15 UTC

If you need I can add or change archive format.

**Comment by dmex** [comment]
> i have used kinda portable version of PH, and have expected that SystemInformer will also support portable mode

@Vasilich

Yes, portable mode is still supported. 

> the maxmind database path seems to disappeared from config

The database path settings were removed.

> SI sees this DB when mmdb file is located in root of SI folder (on same level with SI executable file) though.
Was it planned change or regression?

Yes, there are only two valid locations for the database:

* The current directory (portable mode).
* The `%AppData%\SystemInformer` directory.

You don't need to configure any settings. Just copy `GeoLite2-Country.mmdb` into the same folder as `SystemInformer.exe` or copy into the `%AppData%\SystemInformer` folder.

**Comment by dmex** [comment]
I've updated the nightly builds with options for configuring the geoip keys:

![image](https://user-images.githubusercontent.com/1306177/203634676-dbc38cf6-4536-49c4-a0fd-2ba253cbb9c7.png)

You'll be able to download the geoip database updates after adding your keys 👍

**Comment by Almighty188** [comment]
> I've updated the nightly builds with options for configuring the geoip keys:
> 
> ![image](https://user-images.githubusercontent.com/1306177/203634676-dbc38cf6-4536-49c4-a0fd-2ba253cbb9c7.png)
> 
> You'll be able to download the geoip database updates after adding your keys 👍

Have a question, when generating keys, what options need to be selected as these are the questions it asks:
Old versions of our GeoIP Update program use a different license key format. Will this key be used for [GeoIP Update](https://dev.maxmind.com/geoip/updating-databases?lang=en)? Yes or No
If Yes is selected, which is the correct option:
Generate a license key and config file for use with geoipupdate version 3.1.1 or newer.
Generate a license key and config file for use with geoipupdate versions older than 3.1.1.
This key will be stored in an unhashed format.

Thanks!

**Comment by Vasilich** [comment]
you need new key format, so the answer for question above is "No".

**Comment by Almighty188** [comment]
@dmex - It seems like with the options, entering the MaxMind GeoLite license key, then clicking on Save and then clicking on Download, it will say Invalid key as it seems you have to close the entire Options before the Download will use the key.

**Comment by dmex** [comment]
> clicking on Save and then clicking on Download, it will say Invalid key as it seems you have to close the entire Options

The key is used when you click Save. Closing the options window doesn't do anything.

**Comment by Almighty188** [comment]
> > clicking on Save and then clicking on Download, it will say Invalid key as it seems you have to close the entire Options
> 
> The key is used when you click Save. Closing the options window doesn't do anything.

Yes, as I mentioned.  I entered the key, clicked on Save first and then Download but it says invalid key and did not work until I closed the options window and reopened the options window to go to the same section and clicked on Download.  Basically saying the Download on the options doesn't recognize the new key after saving unless the options window is closed first before using the download in the options window.

**Comment by SMooxx** [comment]
SO where do we need to put this geolites database file?

**Comment by Almighty188** [comment]
Does anyone know how often the GeoIP database is updated?

**Comment by dmex** [comment]
@Almighty188 

The update information is on the maxmind status page:

```
Database | Update Schedule:

Every Tuesday and Friday.
```

**Comment by Almighty188** [comment]
> @Almighty188
> 
> The update information is on the maxmind status page:
> 
> ```
> Database | Update Schedule:
> 
> Every Tuesday and Friday.
> ```

Many thanks @dmex.  I tried looking on maxmind status page at:
https://status.maxmind.com/ and https://status.maxmind.com/pages/history/53fcfbb2ac0c957972000235 which is the history but couldn't find the update information, can you post the URL where you found it as I couldn't find it when I tried even after your response above.

**Comment by pasha-zzz** [comment]
> can you post the URL where you found it as I couldn't find it when I tried even after your response above.

https://support.maxmind.com/hc/en-us/articles/4408216129947-Download-and-Update-Databases#h_01G3XX402XKD3J1CMWKNKMDYYZ

**Comment by Almighty188** [comment]
Many thanks @pasha-zzz
